src/main/java/org/testng/internal/ConfigInvoker.java

```diff
@@ -189,45 +195,35 @@ public void invokeAfterGroupsConfigurations(
     ITestNGMethod[] afterMethodsArray = afterMethods.keySet().toArray(new ITestNGMethod[0]);
     // don't pass the IClass or the instance as the method may be external
     // the invocation must be similar to @BeforeTest/@BeforeSuite
-    invokeConfigurations(
-        null, afterMethodsArray, suite, params, /* no parameter values */ null, instance);
+    ConfigMethodAttributes attributes = new Builder()
+        .usingConfigMethodsAs(afterMethodsArray)
+        .forSuite(suite)
+        .usingParameters(params)
+        .usingInstance(instance)
+        .build();
+
+    invokeConfigurations(attributes);
 
     // Remove the groups so they don't get run again
     groupMethods.removeAfterGroups(filteredGroups.keySet());
   }
 
   public void invokeConfigurations(
-      IClass testClass,
-      ITestNGMethod[] allMethods,
-      XmlSuite suite,
-      Map<String, String> params,
-      Object[] parameterValues,
-      Object instance) {
-    invokeConfigurations(
-        testClass, null, allMethods, suite, params, parameterValues, instance, null);
-  }
-
-  public void invokeConfigurations(
-      IClass testClass,
-      ITestNGMethod currentTestMethod,
-      ITestNGMethod[] allMethods,
-      XmlSuite suite,
-      Map<String, String> params,
-      Object[] parameterValues,
-      Object instance,
-      ITestResult testMethodResult) {
-    if (null == allMethods || allMethods.length == 0) {
+      ConfigMethodAttributes configMethodAttributes) {
+    if (null == configMethodAttributes.getConfigMethods()
```

Comment:
Any way to enforce that configMethodAttributes.getConfigMethods() will never return null?

Reply:
I can check that. But I don't know the impact. So need more time to get to that. I will check that in a separate PR.

Reply:
Ok.. I just added logic to initialize it to an empty array if it turns out to be null.